src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer.cs

```diff
@@ -46,33 +46,38 @@ private void AnalyzeArrayCreationExpression(SyntaxNodeAnalysisContext context, I
             return;
 
         // Analyze the statements that follow to see if they can initialize this array.
-        var matches = TryGetMatches(semanticModel, arrayCreationExpression, expressionType, cancellationToken);
+        var allowInterfaceConversion = context.GetAnalyzerOptions().PreferCollectionExpressionForInterfaces.Value;
+        var matches = TryGetMatches(semanticModel, arrayCreationExpression, expressionType, allowInterfaceConversion, cancellationToken, out var changesSemantics);
```

Comment:
most of the change is getting hte info about if sematnics changed, passing it into the diagnostic, and presenting it in the fix.